src/mlpack/bindings/go/print_doc_functions_impl.hpp

```diff
@@ -0,0 +1,544 @@
+  if (input != "")
+    result = result + input + ", ";
+  oss.str(""); // Reset it.
+  result = result + "param";
+  result = result + ")";
+  result = util::HyphenateString(result, 4);
+
+  return result;
+}
+
+/**
+ * Given the name of a model, print it.  Here we do not need to modify anything.
+ */
+inline std::string PrintModel(const std::string& modelName)
+{
+  std::string goStrippedType, strippedType, printedType, defaultsType;
+  StripType(modelName, goStrippedType, strippedType, printedType, defaultsType);
+
+  return strippedType;
+}
+
+/**
+ * Given the name of a matrix, print it.  Here we do not need to modify
+ * anything.
+ */
+inline std::string PrintDataset(const std::string& datasetName)
+{
+  return datasetName;
+}
+
+/**
+ * Given the name of a binding, print its invocation.
+ */
+inline std::string ProgramCall(const std::string& programName)
+{
+  std::ostringstream oss;
+  std::string goProgramName = CamelCase(programName, false);
+
+  // Determine if we have any output options.
+  const std::map<std::string, util::ParamData>& parameters = CLI::Parameters();
+  oss << "    // Initialize optional parameters for " << goProgramName << "()."
+      << "\n";
+  oss << "param := mlpack." << goProgramName << "Options()\n";
+
+  std::vector<std::string> outputOptions;
+  for (auto it = CLI::Parameters().begin(); it != CLI::Parameters().end(); ++it)
+  {
+    const util::ParamData& d = it->second;
+    if (!d.input)
+      outputOptions.push_back(it->first);
+  }
+  std::string result = oss.str();
+  oss.str("");
+  std::string param = "";
+  // Now iterate over every optional input option.
+  for (auto it = parameters.begin(); it != parameters.end(); ++it)
+  {
+    if (it->second.input && !it->second.required && !it->second.persistent)
+    {
+      // Print the input option.
+      oss << "param." << CamelCase(it->second.name, false) << " = ";
+      std::string value;
+      CLI::GetSingleton().functionMap[it->second.tname]["DefaultParam"](
+          it->second, NULL, (void*) &value);
+      oss << value;
+      oss << "\n";
+      param = oss.str();
+    }
+  }
+
+  oss << "\n";
+  result += oss.str();
+  oss.str("");
+  oss << result;
+
+  // Now print output lines.
+  size_t outputs = 0;
+  for (auto it = parameters.begin(); it != parameters.end(); ++it)
+  {
+    if (!it->second.input)
+    {
+      if (outputs > 0)
+      {
+        oss << ", ";
+        oss << it->second.name;
+      }
+      else
+      {
+        oss << it->second.name;
+      }
+      ++outputs;
+    }
+  }
+
+  oss << " := mlpack." << goProgramName << "(";
+  for (auto i = parameters.begin(); i != parameters.end(); ++i)
+  {
+    if (i->second.input && i->second.required && i != parameters.end())
+      oss << CamelCase(i->second.name, true) << ", ";
+    else if (i == parameters.end())
+      oss << CamelCase(i->second.name, true);
+  }
+  if (param != "")
+    oss << "param";
+  oss << ")";
+
+  result = util::HyphenateString(oss.str(), 4);
```

Comment:
Same here, maybe we need to call `HyphenateString` on each individual line that comes back?  We currently have this output:

```
import (
  "mlpack/build/src/mlpack/bindings/go/mlpack"
  "gonum.org/v1/gonum/mat"
)

    // Initialize optional parameters for ApproxKfn().
    param := mlpack.ApproxKfnOptions()
    param.Algorithm = "ds"
    param.CalculateError = false
    param.ExactDistances = mat.NewDense(1, 1, nil)
    param.InputModel = nil
    param.K = 0
    param.NumProjections = 5
    param.NumTables = 5
    param.Query = mat.NewDense(1, 1, nil)
    param.Reference = mat.NewDense(1, 1, nil)
    
    distances, neighbors, output_model := mlpack.ApproxKfn(param)
```

but we want this:

```
import (
  "mlpack/build/src/mlpack/bindings/go/mlpack"
  "gonum.org/v1/gonum/mat"
)

// Initialize optional parameters for ApproxKfn().
param := mlpack.ApproxKfnOptions()
param.Algorithm = "ds"
param.CalculateError = false
param.ExactDistances = mat.NewDense(1, 1, nil)
param.InputModel = nil
param.K = 0
param.NumProjections = 5
param.NumTables = 5
param.Query = mat.NewDense(1, 1, nil)
param.Reference = mat.NewDense(1, 1, nil)

distances, neighbors, output_model := mlpack.ApproxKfn(param)
```

Reply:
Updated this part, Calling `HyphenateString` on each individual comes back works.
Thank You. :)